File: 392-is-subsequence/392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == '' and t == '':
            return True
        
        p = 0 # pointer for s
        q = 0 # pointer for t
        
        for i in range(len(t)):
            
            if p == len(s):
                return True
            
            if s[p] == t[i]:
                p += 1
                
        if p == len(s):
            return True
        return False
            
        
            
        

File: 392-is-subsequence/392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        
        int p = 0; // Pointer for string s
        int q = 0; // Pointer for string t
        
        if(s == "" && t == "")
            return true;
        
        for(int i = 0; i < t.size(); i++)
        {
            if(s[p] == t[q])
            {
                p++;
                q++;
            }
            else
            {
                q++;
            }
            
            if(p == s.size())
                return true;
        }
        return false;
    }
};